text/0000-infered-types.md

```diff
@@ -0,0 +1,149 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When crating a struct or enum, infered types can simplify the type into just a underscore. It is important to note, however, that they do not work when the type is not specific enough to be infered like: type parameters. Below are some examples where they do and don't work.
+
+Function calls (structs):
+```rust
+fn my_function(data: MyStruct) { /* ... */ }
+
+// my_function(MyStruct {
+//     value: 1
+// });
+my_function(_ {
+   value: 1
+});
+```
+
+Function calls (impl):
+```rust
+fn my_function(data: MyStruct) { /* ... */ }
+
+// my_function(MyStruct::new()});
+my_function(_::new());
+```
+
+Function returns (enum):
+```rust
+fn my_function() -> MyEnum {
+   // MyEnum::MyVarient
+   _::MyVarient
+}
+```
+
+Match arms:
+```rust
+enum Example {
+   One,
+   Two
+}
+
+fn my_fn(my_enum: Example) -> String {
+   match my_enum {
+      _::One => "One!",
+      _::Two => "Two!"
+   }
+}
+```
+
+It is important to note that `_` only represents the type; if you have (for example) another enum that can be coerced into the type, you will need to specify it manually. Additionally, any traits required to call an impl will still have to be imported.
+
+```rust
+fn my_function(data: MyStruct) { /* ... */ }
+
+
+my_function(MyStruct2::do_something().into()); // ✅
+
+
+my_function(_::do_something().into()); // ❌ error[E0599]: variant or associated item not found in `MyStruct`
+```
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The underscore operator infers the type of a stuct or enum where there is enough type information in a context in order to infer an exact type. In a statement like `_::Variant`, you can imagine the underscore to be the type. That means that anything you could do with an actual type like `MyEnum::Variant` will still apply in an infered enum. Ultimately, the enum or struct doesn't need to be imported but, traits and other specfied things will need to be imported.
```

Comment:
I'm wondering if the enum use case is necessary, when the alternative is:
```rust
use MyEnum::*;
```

Language servers and similar tools will auto-suggest these kinds of imports for most developers.